tools/src/Tools_Docgen.res

```diff
@@ -38,21 +38,21 @@ type rec item =
       /** Additional documentation for constructors and record fields, if available. */
       detail?: detail,
     })
-  | @as("module") Module(docsForModule)
+  | @as("module")
```

Comment:
The decorator `as("module")` create a filed `kind: "module"`.

But when used in pattern matching the `Module` variant the generated code is:

```rescript
if (docItem.TAG === "Module")
```